pkg/vault/util.go

```diff
@@ -0,0 +1,79 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// Login TODO
+func Login(vaultClient VaultType, vaultConfig *Config) error {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	avpConfigPath := filepath.Join(home, ".avp", "config.json")
+	if _, err := os.Stat(avpConfigPath); err == nil {
+		// Open our jsonFile
+		jsonFile, err := os.Open(avpConfigPath)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// defer the closing of our jsonFile so that we can parse it later on
+		defer jsonFile.Close()
+
+		byteValue, _ := ioutil.ReadAll(jsonFile)
+
+		var result map[string]interface{}
+		json.Unmarshal([]byte(byteValue), &result)
+
+		vaultConfig.VaultAPIClient.SetToken(result["vault_token"].(string))
+		tokenself, _ := vaultConfig.VaultAPIClient.Auth().Token().LookupSelf()
+		expireTime := tokenself.Data["expire_time"]
+		today := time.Now()
+		myDate, _ := time.Parse("2021-01-08T08:21:27.195176037Z", expireTime.(string))
+
+		if today.Before(myDate) {
+			err = vaultClient.Login()
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		err = vaultClient.Login()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// SetToken TODO
+func SetToken(client *Client, token string) error {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	path := filepath.Join(home, ".avp")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		os.Mkdir(path, 0755)
+	}
+
+	viper.Set("VAULT_TOKEN", token)
+	err = viper.WriteConfigAs(filepath.Join(path, "config.json"))
```

Comment:
Interesting, we use Viper to store and write the token to a file but then we don't use Viper to read it back, I guess so that we don't change any other Viper parameters?

Reply:
Yeah, i was thinking reading that file in would override. I could change to not use viper to write the file for the sake of continuity

Reply:
I guess that would make sense since we're not reading with Viper

Reply:
You could also setup another instance of Viper to make reading/writing easier

Reply:
> You could also setup another instance of Viper to make reading/writing easier

Didn't see a way to do this (Or at least i didn't comprehend how to do that)

Reply:
https://github.com/spf13/viper#working-with-multiple-vipers i guess i missed this